client-course-schedulizer/src/utilities/helpers/caseFunctions.ts

```diff
@@ -349,6 +350,6 @@ export const durationCase = (value: string): number => {
   return endTimeMoment.diff(startTimeMoment, "minutes");
 };
 
-export const yearCase = (value: string): number | string => {
-  return Number.isInteger(Number(value)) ? Number(value) : value;
+export const yearCase = (value: string): number | string | undefined => {
+  return Number.isInteger(Number(value)) && value ? Number(value) : value;
```

Comment:
Does this work? if `value` is falsy, then it will short circuit and not try to cast to a number. Added parens to make it more explicit what the precondition is for the ternary operator. 

```suggestion
  return (value && Number.isInteger(Number(value))) ? Number(value) : value;
```